front/src/modules/infra/components/InfraSelector/InfraSelectorModalBodyStandard.tsx

```diff
@@ -49,6 +54,20 @@ export default function InfraSelectorModalBodyStandard({
   const { closeModal } = useContext(ModalContext);
   const navigate = useNavigate();
 
+  const { data: subjectsList } = mockedEditoastApi.endpoints.getSubjects.useQuery();
+
+  const { data: resourceGrants } =
+    mockedEditoastApi.endpoints.getGrantsByResourceType.useQuery('infra');
+
+  const payload = useMemo(() => ({ infra: infrasList.map((infra) => infra.id) }), [infrasList]);
+
+  const userResourcesGrants = useResourcesGrants(payload);
+
+  const userGrantByInfra = useMemo(
+    () => keyBy(userResourcesGrants?.infra, 'id'),
+    [userResourcesGrants]
+  );
```

Comment:
If I understand correctly, the grants of the current user are re-fetched each time the infras are filtered :/

I think it would be better to handle this only when the infras are fetched when the component is mounted.